### PR TITLE
[HIVEMALL-5-pre] Java code formatter 

### PR DESCRIPTION
**Pull request by takuti**
## What changes were proposed in this pull request?

In order to make contribution to Hivemall more easier, we need to prepare contribution guide as soon as possible. This PR is the first step toward publishing a complete document on https://hivemall.incubator.apache.org

For now, I added (temporal) contribution guide to README and require contributors to follow specific cording conventions hereafter.

## What type of PR is it?

Documentation?

## What is the Jira issue?

https://issues.apache.org/jira/browse/HIVEMALL-5

---

Originally, `eclipse-style.xml` was created based on google's coding conventions. However, in this PR, I suggest to follow coding conventions of Hive hereafter; since Hivemall clearly stands on Hive, following its conventions should be the most reasonable choice.

@myui If you strongly want to use different kind of conventions, I will update this PR so.

### CODE REVIEW

File: core/src/main/java/hivemall/anomaly/SDAR2D.java [hunk 1]
```diff
@@ -33,129 +33,123 @@
...
 
-    /**
-     * @param x series of input in LIFO order
-     * @param k AR window size
-     * @return x_hat predicted x
-     * @link https://en.wikipedia.org/wiki/Matrix_multiplication#Outer_product
+    // solve A in the following Yule-Walker equation
+    // C_j = ∑_{i=1}^{k} A_i C_{j-i} where j = 1..k, C_{-i} = C_i'
+    /*
+     * /C_1\ /A_1\ /C_0 |C_1' |C_2' | . . . |C_{k-1}' \ |---| |---| |--------+--------+--------+
```
Comment (myui) [inline core/src/main/java/hivemall/anomaly/SDAR2D.java]: This comment should not be formatted.

File: core/src/main/java/hivemall/classifier/AROWClassifierUDTF.java [hunk 2]
```diff
@@ -41,172 +41,165 @@
  *     In Proc. NIPS, 2009.
  * </pre>
  */
-@Description(
-        name = "train_arow",
-        value = "_FUNC_(list<string|int|bigint> features, int label [, const string options])"
-                + " - Returns a relation consists of <string|int|bigint feature, float weight, float covar>",
-        extended = "Build a prediction model by Adaptive Regularization of Weight Vectors (AROW) binary classifier")
+@Description(name = "train_arow", value = "_FUNC_(list<string|int|bigint> features, int label [, const string options])"
+    + " - Returns a relation consists of <string|int|bigint feature, float weight, float covar>", extended = "Build a prediction model by Adaptive Regularization of Weight Vectors (AROW) binary classifier")
```
Comment (myui) [inline core/src/main/java/hivemall/classifier/AROWClassifierUDTF.java]: too long line.

File: core/src/main/java/hivemall/anomaly/SingularSpectrumTransform.java [hunk 3]
```diff
@@ -41,161 +41,160 @@
...
+
+      // create current trajectory matrix and find its left singular vectors
+      RealMatrix G = new Array2DRowRealMatrix(window, nCurrentWindow);
+      int currentHead = pastSize + currentOffset;
+      for (int i = 0; i < nCurrentWindow; i++) {
+        G.setColumn(i, Arrays.copyOfRange(xSeries, currentHead + i, currentHead + i + window));
+      }
+
+      switch (scoreFunc) {
+      case svd:
```
Comment (myui) [inline core/src/main/java/hivemall/anomaly/SingularSpectrumTransform.java]: `case` better to be indented as seen in Google coding style.

File: core/src/main/java/hivemall/classifier/KPAPredictUDAF.java [hunk 4]
```diff
@@ -40,189 +40,187 @@
...
-                this.w3OI = HiveUtils.asNumberOI(parameters[5]);
-            }
+    @Nullable
+    private transient PrimitiveObjectInspector xhOI, xkOI;
+    @Nullable
+    private transient PrimitiveObjectInspector w0OI, w1OI, w2OI, w3OI;
 
-            return PrimitiveObjectInspectorFactory.writableDoubleObjectInspector;
-        }
+    public Evaluator() {
```
Comment (myui) [inline core/src/main/java/hivemall/classifier/KPAPredictUDAF.java]: `public Evaluator() {}` is preferred as seen in Google coding style.
https://google.github.io/styleguide/javaguide.html#s4.1.3-braces-empty-blocks

File: core/src/main/java/hivemall/utils/lang/Preconditions.java [hunk 5]
```diff
@@ -26,16 +26,17 @@
 
 public final class Preconditions {
 
-    private Preconditions() {}
+  private Preconditions() {
```
Comment (myui) [inline core/src/main/java/hivemall/utils/lang/Preconditions.java]: `private Preconditions() {}` is preferred.

**Comment by myui** [comment]
I need to rewrite [JIRA issue](https://issues.apache.org/jira/browse/HIVEMALL-5). Let's discuss on the next Monday (or Tues).

**Comment by takuti** [comment]
@myui Could you share how you are planning to re-write the JIRA issue here If possible? Keeping our discussion in public is generally better :)

Overall, I suspect that you want to follow the google coding convention rather than Hive's one. Is it correct? I'm personally okay for any conventions as long as they are well-documented. If we define some exceptions on either google / Hive (oracle) conventions, we manually need to document them ✍️

**Comment by myui** [comment]
>  I suspect that you want to follow the google coding convention rather than Hive's one. 

Right.

But, completely following Google coding style has problems in formatting existing codes. Manual comment rewriting and string concatenation rewriting is required.

File: core/src/main/java/hivemall/anomaly/SDAR2D.java [hunk 6]
```diff
@@ -33,129 +33,123 @@
...
+   * @param x series of input in LIFO order
+   * @param k AR window size
+   * @return x_hat predicted x
+   * @link https://en.wikipedia.org/wiki/Matrix_multiplication#Outer_product
+   */
+  @Nonnull
+  public RealVector update(@Nonnull final ArrayRealVector[] x, final int k) {
+    Preconditions.checkArgument(x.length >= 1, "x.length MUST be greather than 1: " + x.length);
+    Preconditions.checkArgument(k >= 0, "k MUST be greather than or equals to 0: ", k);
+    Preconditions.checkArgument(k < _C.length, "k MUST be less than |C| but " + "k=" + k + ", |C|="
```
Comment (myui) [inline core/src/main/java/hivemall/anomaly/SDAR2D.java]: Here, you can find String concatenation problems.

**Comment by myui** [comment]
Current `eclipse-style.xml` is using modified version of Sun's coding style (no comment formatting), not google one. It's better to evaluate `Google coding style + no comment formatting` though. 

Google coding style provides both Eclipse format and IntelliJ format but different formatting is applied due to mismatch in functionality of IDE. So, providing only Eclipse format (which can be loaded in IntelliJ) is preferred.

File: core/src/main/java/hivemall/anomaly/SDAR2D.java (hunk 6 above)
Comment (takuti) [inline core/src/main/java/hivemall/anomaly/SDAR2D.java]: @myui Oh, it's huge...I understand what you are afraid ;(

**Comment by takuti** [comment]
@myui Thank you for sharing your opinion! I realized that this issue requires us to undergo significant amount of discussion and modification of the current code.

So, I renamed the title of this PR to **[HIVEMALL-5-pre]** as the first step to publish our fixed version of coding guideline. There is no update in terms of coding conventions, and I just applied the formatter to the current code and updated README for future contributors.

If it's acceptable for you, I like to merge this soon :)

File: core/src/main/java/hivemall/anomaly/SingularSpectrumTransform.java [hunk 7]
```diff
@@ -165,8 +165,8 @@ private double computeScoreSVD(@Nonnull final RealMatrix H, @Nonnull final RealM
     /**
      * Implicit Krylov Approximation (IKA) based naive scoring.
      *
-     * Number of iterations for the Power method and QR method is fixed to 1 for efficiency. This
-     * may cause failure (i.e. meaningless scores) depending on datasets and initial values.
+     * Number of iterations for the Power method and QR method is fixed to 1 for efficiency. This may cause failure (i.e. meaningless scores)
```
Comment (myui) [inline core/src/main/java/hivemall/anomaly/SingularSpectrumTransform.java]: Better to set Javadoc comment formatting off for the time being.

File: core/src/main/java/hivemall/ftvec/binning/NumericHistogram.java [hunk 8]
```diff
@@ -30,11 +30,9 @@
...
  * **THIS CLASS IS IMPORTED FROM HIVE 2.1.0 FOR COMPATIBILITY**
  *
- * A generic, re-usable histogram class that supports partial aggregations. The algorithm is a
- * heuristic adapted from the following paper: Yael Ben-Haim and Elad Tom-Tov,
- * "A streaming parallel decision tree algorithm", J. Machine Learning Research 11 (2010), pp.
- * 849--872. Although there are no approximation guarantees, it appears to work well with adequate
- * data and a large (e.g., 20-80) number of histogram bins.
+ * A generic, re-usable histogram class that supports partial aggregations. The algorithm is a heuristic adapted from the following paper: Yael
+ * Ben-Haim and Elad Tom-Tov, "A streaming parallel decision tree algorithm", J. Machine Learning Research 11 (2010), pp. 849--872. Although there are
+ * no approximation guarantees, it appears to work well with adequate data and a large (e.g., 20-80) number of histogram bins.
```
Comment (myui) [inline core/src/main/java/hivemall/ftvec/binning/NumericHistogram.java]: same as here.

**Comment by myui** [comment]
Merged with some modifications. 100 spaces for Javadoc comment, as same as Google style, works fine.

**Pull request closed without merging**